### PR TITLE
Transform f1 othr reg liab

### PR DESCRIPTION
**Pull request by zschira**
# Overview
This PR implements transformations for the `f1_othr_reg_liab` table. This table contains a `starting_balance`, `ending_balance`, deltas, and a description and list of account numbers. The description and accounts are very freeform, making it difficult to apply a nice structure to this table. Outside of the unstructured nature of this table, it is very straightforward. All I have done is renaming, unstacking starting/ending balances, and dropping invalid rows.

# PR Checklist

- [x] Merge the most recent version of the branch you are merging into (probably `dev`).
- [x] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [x] Make sure you've included good docstrings.

### CODE REVIEW

File: src/pudl/metadata/fields.py [hunk 1]
```diff
@@ -368,6 +368,15 @@
         "type": "string",
         "description": "Geodetic coordinate system identifier (e.g. NAD27, NAD83, or WGS84).",
     },
+    "debits_account_credited": {
+        "type": "string",
+        "description": "Description of the account number credited from making debit adjustment to other regulatory liabilities.",
```
Comment (cmgosnell) [inline src/pudl/metadata/fields.py:373]: i know this is accounting speak and is actually understandable/consistent with accounting speak but barf on the credit debit credits.

Comment (cmgosnell) [reply src/pudl/metadata/fields.py:373]: could we call this column just `account_number` or `account_detail` or something? or `account_credited`? it looks like one or multiple actual account numbers or some silly string like `Various` or `footnote`.

**Review by cmgosnell — CHANGES_REQUESTED** [review]
one request to rename a column but everything else looks grand (at least grand enough given the messiness of the underlying data!)

File: src/pudl/metadata/fields.py (hunk 1 above)
Comment (zschira) [reply src/pudl/metadata/fields.py:373]: Yeah I didn't like this name at all either, and the credit/debit detail is available in the description, so this seems like a good change to me

**Pull request merged**